File: coslu-reader/samples/sample.cs
// Coslu Reader — sample C# (Tier S via Shiki)
using System;
using System.Linq;

namespace CosluLabz;

public record Familia(string Nome, int Tier = 1);

public static class Program
{
    public static void Main(string[] args)
    {
        var fams = new[] { new Familia("texto"), new Familia("hex", 0) };
        foreach (var f in fams.Where(x => x.Tier >= 0))
            Console.WriteLine($"{f.Nome} → tier {f.Tier}");
    }
}

File: coslu-reader/samples/sample.js
// Coslu Reader — sample JavaScript (Tier S via Shiki)
const familias = ["texto", "dados", "hex"];

async function carrega(nome, { tier = 1 } = {}) {
  const t0 = performance.now();
  await new Promise((r) => setTimeout(r, 10));
  return `${nome} (tier ${tier}) em ${(performance.now() - t0).toFixed(1)}ms`;
}

for (const f of familias) {
  carrega(f, { tier: f === "hex" ? 0 : 1 }).then(console.log);
}

export { carrega };
